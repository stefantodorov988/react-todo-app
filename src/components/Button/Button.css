.button {
    background: none;
    border: none;
    outline: none;
    color: rgb(199, 199, 199);
    cursor: pointer;
  }
  
  .button:hover {
    color: black;
  }
  
  .button.selected {
    color: rgb(72, 128, 255);
  }