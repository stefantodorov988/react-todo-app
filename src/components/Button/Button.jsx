import "./Button.css";

export const Button = ({handleClick, children, selected}) => {
    const classes = children === selected ? "button selected" : "button";

    return (
        <button className={classes} onClick={handleClick}>{children}</button>
    );
}