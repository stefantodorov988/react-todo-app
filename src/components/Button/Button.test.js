import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Button } from "./Button";

test("renders button with correct text", () => {
  const handleClick = jest.fn(); // Create a mock function for the handleClick prop
  const selected = "Button 1";
  const buttonText = "Button 1";

  const { getByText } = render(
    <Button handleClick={handleClick} selected={selected}>
      {buttonText}
    </Button>
  );

  const button = getByText(buttonText);
  expect(button).toBeInTheDocument();
});

test("calls handleClick when button is clicked", () => {
  const handleClick = jest.fn(); // Create a mock function for the handleClick prop
  const selected = "Button 1";
  const buttonText = "Button 1";

  const { getByText } = render(
    <Button handleClick={handleClick} selected={selected}>
      {buttonText}
    </Button>
  );

  const button = getByText(buttonText);
  fireEvent.click(button);
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test("renders button with selected class when selected is true", () => {
  const handleClick = jest.fn();
  const selected = "Button 1";
  const buttonText = "Button 1";

  const { getByText } = render(
    <Button handleClick={handleClick} selected={selected}>
      {buttonText}
    </Button>
  );

  const button = getByText(buttonText);
  expect(button).toHaveClass("selected");
});

test("renders Button component correctly", () => {
  const handleClick = jest.fn();
  const selected = "Button 1";
  const buttonText = "Button 1";

  const { container } = render(
    <Button handleClick={handleClick} selected={selected}>
      {buttonText}
    </Button>
  );

  expect(container.firstChild).toMatchSnapshot();
});
