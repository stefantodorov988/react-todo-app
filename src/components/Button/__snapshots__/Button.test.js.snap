// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component correctly 1`] = `
<button
  class="button selected"
>
  Button 1
</button>
`;
